lib/update-linkstatus-obj.js

```diff
@@ -7,12 +7,21 @@
  * @param {Array} linkStatus - The existing link status to update.
  * Each status is an object with properties `link`, `status`, `status_code`, `line_number`, `position`, `error_message`, `title`, and `children`.
  *
+ * @param {Object} config - The configuration object.
+ *
  * @returns {Array} The updated link status. Each status is an object with properties `link`, `status`, `status_code`, `line_number`, `position`, `error_message`, `title`, and `children`.
  * The returned array is sorted by line number and start column in ascending order.
  */
 'use strict'
 
-function updateLinkStatusObj(astNodes, linkStatus) {
+function updateLinkStatusObj(astNodes, linkStatus, config) {
+  const isJsonFile =
```

Comment:
actually more ideally this wouldn't use the config for this. it should instead be determined per each file